src/java.base/share/classes/java/util/HexDigits.java

```diff
@@ -103,16 +103,20 @@ private HexDigits() {
     }
 
     /**
-     * Combine two hex shorts into one int based on big endian
+     * Return a big-endian packed integer for the 4 ASCII bytes for an input unsigned 2-byte integer.
+     * {@code b0} is the most significant byte and {@code b1} is the least significant byte.
+     * The integer is passed byte-wise to allow reordering of execution.
```

Comment:
The more natural API would be to pass the 16-bit value as an int (or short). Simplifying the callers.
Is there a significant performance difference?
The extraction of the 8-bit values could be done here instead of the caller and still allow hardware specific instruction optimization.

Also below, pass the 32-bit value as an int.

Reply:
> Is there a significant performance difference?

Yes, see https://github.com/openjdk/jdk/pull/14745#issuecomment-1615435719 for the revision at https://github.com/openjdk/jdk/pull/14745/commits/425559f65a3e5f72323abdadbbf9df426e196d9b
On my device, the UUIDBench.toString has a smaller difference of about 8%, but it is still noticeable.

Unfortunately, I don't know architecture well enough to explain this difference.